dotnet/src/Skills/Skills.Web/Bing/BingConnector.cs

```diff
@@ -75,9 +78,17 @@ public async Task<IEnumerable<string>> SearchAsync(string query, int count = 1,
 
         BingSearchResponse? data = JsonSerializer.Deserialize<BingSearchResponse>(json);
 
-        WebPage[]? results = data?.WebPages?.Value;
+        List<string> output = new();
+        if (data?.WebPages?.Value != null)
+        {
+            output.Add(string.Format("Bing Search Results\n\n" +
+                "----\n\n" +
+                $"Page Name: {data?.WebPages?.Value.FirstOrDefault().Name}\n\n" +
+                $"Page Snippet: {data?.WebPages?.Value.FirstOrDefault().Snippet}...\n\n" +
+                $"Page Url: {data?.WebPages?.Value.FirstOrDefault().Url}"));
+        }
 
-        return results == null ? Enumerable.Empty<string>() : results.Select(x => x.Snippet);
+        return output.Count == 0 ? Enumerable.Empty<string>() : output.Take(1);
```

Comment:
Previously we were returning `results.Select(x => x.Snippet)` - all snippets that come in HTTP response, based on `count` property in request query parameter. Now it looks like we will return only first result, which definitely changes the way how this functionality works.

Reply:
Rewritten the method to use generics. Both IEnumerable<string> & IEnumerable<WebPage> types are now supported.

Reply:
> Rewritten the method to use generics. Both `IEnumerable<string>` & `IEnumerable<WebPage>` types are now supported. This allows usage such as,
> 
> ```
>             // Init the bingConnector and import the skill
>             var bingConnector = new BingConnector(_config["Bing_Search_API_Key"], 
>                 uri: "https://soe-azureoai.azure-api.net/bingsearch/v7.0/search?q=");
> 
>             var searchSkill = _sk.ImportSkill(new WebSearchEngineSkill(bingConnector), "bing");
> 
>             ContextVariables contextVars = new($"{text}");
>             contextVars.Set(WebSearchEngineSkill.Parameters.CountParam, $"3");
>             contextVars.Set(WebSearchEngineSkill.Parameters.OffsetParam, $"0");
> 
>             // Perform the search and return the full result object.
>             var search = await _sk.RunAsync(contextVars, searchSkill["GetSearchResults"]);
>             var webPages = JsonConvert.DeserializeObject<List<WebPage>>(search.Result);
> 
>             string results = $"Bing Search Results\n\n" +
>                     "----\n\n";
> 
>             switch(webPages.Count)
>             {
>                 case 0:
>                     results += "No results found.";
>                     break;
>                     case 1:
>                     results += $"Page Name: {webPages[0].Name}\n\n" +
>                         $"Page Snippet: {webPages[0].Snippet}...\n\n" +
>                         $"Page Url: {webPages[0].Url}";
>                     break;
>                 default:
>                     for (int i = 0; i < webPages.Count; i++)
>                     {
>                         results += $"--- Result {i+1} ---\n\n" +
>                             $"Page Name: {webPages[i].Name}\n\n" +
>                             $"Page Snippet: {webPages[i].Snippet}...\n\n" +
>                             $"Page Url: {webPages[i].Url}";
> 
>                         if (i+1 < webPages.Count)
>                         {
>                             results += $"\n\n";
>                         }
>                     }
>                     break;
>             }
>             
>             //// Perform the search and return snippets.
>             // var search = await _sk.RunAsync(contextVars, searchSkill["GetSearchResults"]);
>             // var results = JsonConvert.DeserializeObject<List<string>>(search.Result);
>             
>             // Process results here
> ```

See previous to related comment.